packages/@aws-cdk/aws-rds/lib/serverless-cluster.ts

```diff
@@ -457,7 +457,7 @@ export class ServerlessCluster extends ServerlessClusterBase {
       vpcSecurityGroupIds: securityGroups.map(sg => sg.securityGroupId),
     });
 
-    this.clusterIdentifier = cluster.ref;
+    this.clusterIdentifier = cluster.dbClusterIdentifier?.toLowerCase() || cluster.ref;
```

Comment:
@skinny85 
Thanks for the detailed remarks!!! (And I'm sorry for lack of my consideration.) I understand it well. 

In the future, it would be better to reduce the number of confusing properties to improve usability, so I think method 1 is better.
I'll try to fix it by using feature-flag.